flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/metrics/MetricFetcherImpl.java

```diff
@@ -70,6 +71,8 @@
 
     private long lastUpdateTime;
 
+    private CompletableFuture<Void> fetchMetricsFuture = CompletableFuture.completedFuture(null);
```

Comment:
`@GuardedBy("this")`

Reply:
Use `FutureUtils.completedVoidFuture()` to avoid frequent creating new objects. Same for other occurences.